clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/KafkaStoreIngestionService.java

```diff
@@ -1194,6 +1196,20 @@ public List<ByteBuffer> getPartitionOffsetRecords(String topicName, int partitio
     return offsetRecordArray;
   }
 
+  /**
+   * Updates offset metadata and sync to storage for specified topic partition.
+   * This method is invoked only when isolated ingestion process is reporting topic partition completion to make sure
+   * ingestion process is persisted.
+   */
+  public void syncTopicPartitionOffset(String topicName, int partition) {
```

Comment:
This function is correct and needs no change in this PR, though I had to spend a minute or two staring at it because the for loop condition wigged me out.  We should sometime look at lumping all this math into AmplificationUtils or some such utility class.